### PR TITLE
chore(e2e): add `input-number.spec.ts`

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+            for (const [i, input] of inputs.entries()) {
+                await input.clear();
+                await input.focus();
+                await expect(example).toHaveScreenshot(`06-input-number-${i}.png`);
+            }
+        });
+    });
+
+    test.describe('Caret navigation', () => {
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:85]: The same https://github.com/taiga-family/taiga-ui/pull/6441#discussion_r1450413595

Comment (splincode) [reply projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:85]: fixed

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+
+test.describe('InputNumber', () => {
+    let example: Locator;
+    let input: Locator;
+
+    test.describe('API', () => {
+        test.beforeEach(({page}) => {
+            example = new TuiDocumentationApiPagePO(page).apiPageExample;
+            input = example.locator(
+                '[automation-id=tui-primitive-textfield__native-input]',
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:21]: `getByTestId`

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+        test.describe('if user tries to erase padded decimal zeroes (decimal="always"), mask triggers caret navigation', () => {
+            test.beforeEach(async ({page}) => {
+                await tuiGoto(
+                    page,
+                    '/components/input-number/API?decimal=always&precision=2',
+                );
+
+                example = new TuiDocumentationApiPagePO(page).apiPageExample;
+                input = example.locator(
+                    '[automation-id="tui-primitive-textfield__native-input"]',
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:95]: `getByTestId`

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+                await expect(input).toHaveJSProperty('selectionEnd', '105,'.length);
+                await expect(example).toHaveScreenshot('08-input-number.png');
+            });
+
+            test('105,|00 => Backspace => 105|,00', async ({page}) => {
+                await page.keyboard.press('ArrowLeft');
+                await page.keyboard.press('ArrowLeft');
+                await page.keyboard.press('Backspace');
+                await expect(input).toHaveJSProperty('selectionStart', '105'.length);
+                await expect(input).toHaveJSProperty('selectionEnd', '10,'.length);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:122]:
```suggestion
                await expect(input).toHaveJSProperty('selectionStart', '105'.length);
                await expect(input).toHaveJSProperty('selectionEnd', '105'.length);
```

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+            });
+        });
+
+        test.describe('if user tries to erase thousand separator, mask triggers caret navigation', () => {
+            test.beforeEach(async ({page}) => {
+                await tuiGoto(page, '/components/input-number/API?decimal=not-zero');
+
+                example = new TuiDocumentationApiPagePO(page).apiPageExample;
+                input = example.locator(
+                    '[automation-id="tui-primitive-textfield__native-input"]',
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:142]: `getByTestId`

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+    });
+
+    test.describe('[min] prop', () => {
+        test.describe('[min] property is positive number', () => {
+            test.beforeEach(async ({page}) => {
+                await tuiGoto(page, '/components/input-number/API?min=5');
+
+                example = new TuiDocumentationApiPagePO(page).apiPageExample;
+                input = example.locator(
+                    '[automation-id="tui-primitive-textfield__native-input"]',
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:177]: `getByTestId `

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+                await expect(example).toHaveScreenshot('25-input-number.png');
+            });
+        });
+    });
+
+    test.describe('value formatting on blur', () => {
+        test.beforeEach(({page}) => {
+            example = new TuiDocumentationApiPagePO(page).apiPageExample;
+            input = example.locator(
+                '[automation-id="tui-primitive-textfield__native-input"]',
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:316]: `getByTestId `

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts (hunk 2 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-number/input-number.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts (hunk 3 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-number/input-number.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts (hunk 4 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-number/input-number.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts (hunk 5 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-number/input-number.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts [hunk 8]
```diff
@@ -0,0 +1,386 @@
...
+            });
+        });
+
+        test.describe('[min] property is negative number', () => {
+            test.beforeEach(async ({page}) => {
+                await tuiGoto(page, '/components/input-number/API?min=-5');
+
+                example = new TuiDocumentationApiPagePO(page).apiPageExample;
+                input = example.locator(
+                    '[automation-id="tui-primitive-textfield__native-input"]',
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-number/input-number.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts [hunk 9]
```diff
@@ -0,0 +1,386 @@
...
+    });
+
+    test.describe('[max] prop', () => {
+        test.describe('[max] property is negative number', () => {
+            test.beforeEach(async ({page}) => {
+                await tuiGoto(page, '/components/input-number/API?max=-5');
+
+                example = new TuiDocumentationApiPagePO(page).apiPageExample;
+                input = example.locator(
+                    '[automation-id="tui-primitive-textfield__native-input"]',
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-number/input-number.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts (hunk 7 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input-number/input-number.spec.ts]: fixed

**Pull request merged**